src/unit.rs

```diff
@@ -2095,36 +2096,47 @@ impl<'input, 'abbrev, 'unit, Endian> EntriesCursor<'input, 'abbrev, 'unit, Endia
     pub fn next_sibling<'me>
         (&'me mut self)
          -> Result<Option<(&'me DebuggingInformationEntry<'input, 'abbrev, 'unit, Endian>)>> {
-        if self.cached_current.is_some() {
-            let sibling_ptr = self.current().unwrap().attr_value(constants::DW_AT_sibling);
-            if let Some(AttributeValue::UnitRef(offset)) = sibling_ptr {
-                if self.unit.is_valid_offset(offset) {
-                    // Fast path: this entry has a DW_AT_sibling
-                    // attribute pointing to its sibling.
-                    self.input = self.unit.range_from(offset..);
-                    self.cached_current = None;
-                    try!(self.next_entry());
-                    return Ok(self.current());
+        if self.current().is_none() {
+            // We're already at the null for the end of the sibling list.
+            return Ok(None);
+        }
+
+        // Loop until we find an entry at the current level.
+        let mut depth = 0;
+        loop {
+            if self.current().map(|entry| entry.has_children()) == Some(true) {
```

Comment:
`blah.unwrap_or(false)` can replace `blah == Some(true)` but I don't feel strongly about it.